### PR TITLE
fix(node): fix version headers

### LINKED ISSUE #28969: @types/node@10.10.1 throws Error: Cannot find name 'Set'
- [x] I tried using the `@types/node@10.10.1` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @SimonSchick 

I have a SDK which does not use @types/node directly. I use `tsc@2.9.2` to validate the definition file (with no tsconfig.json). One of the dependencies (rollup@0.61.2) uses @types/node

Running `tsc play.d.ts` throws an Error: 

```
node_modules/@types/node/index.d.ts:804:38 - error TS2304: Cannot find name 'Set'.

804         allowedNodeEnvironmentFlags: Set<string>;
                                         ~~~
```

This change was introduced in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/28918.

Downgrading to @types/node@10.9.4 fixes this Error.

To reproduce,  clone and checkout this branch https://github.com/onerain88/Play-SDK-JS/tree/fix_disconnect, then run `npm run lint`。

### LINKED ISSUE #29172: error TS2304: Cannot find name 'Set'.  815 allowedNodeEnvironmentFlags: Set<string>;
```
D:\workshop\www
λ node -v
v8.12.0

D:\workshop\www
λ npm -v
6.4.1

D:\workshop\www
λ npm install typescript -g
C:\Users\dell\AppData\Roaming\npm\tsc -> C:\Users\dell\AppData\Roaming\npm\node_modules\typescript\bin\tsc
C:\Users\dell\AppData\Roaming\npm\tsserver -> C:\Users\dell\AppData\Roaming\npm\node_modules\typescript\bin\tsserver
+ typescript@3.0.3
updated 1 package in 3.061s

D:\workshop\www
λ tsc --version
Version 3.0.3

D:\workshop\www
λ mkdir typescript_learning

D:\workshop\www
λ cd typescript_learning

D:\workshop\www\typescript_learning
λ npm init -y
Wrote to D:\workshop\www\typescript_learning\package.json:

{
  "name": "typescript_learning",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}



D:\workshop\www\typescript_learning  (typescript_learning@1.0.0)
λ tsc --init
message TS6071: Successfully created a tsconfig.json file.

D:\workshop\www\typescript_learning  (typescript_learning@1.0.0)
λ npm install @types/node --dev-save
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN typescript_learning@1.0.0 No description
npm WARN typescript_learning@1.0.0 No repository field.

+ @types/node@10.11.0
added 1 package from 30 contributors in 1.657s

D:\workshop\www\typescript_learning  (typescript_learning@1.0.0)
λ touch helloworld.ts

D:\workshop\www\typescript_learning  (typescript_learning@1.0.0)
λ tsc helloworld.ts
node_modules/@types/node/index.d.ts:815:38 - error TS2304: Cannot find name 'Set'.

815         allowedNodeEnvironmentFlags: Set<string>;
```
helloworld.ts
```
var a:string = "HelloWorld"
console.log(a)
```

What should I do?

### PR DESCRIPTION
**Pull request by SimonSchick**
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

@andy-ms @anthonyvercolano

Potential workaround for: https://github.com/Microsoft/types-publisher/issues/494

This PR might be redundant if this is fixed in the publisher instead.

I suggest you should add a check that causes any typings with the `dt-header` rule disabled to fail in the future 😄 

Fixes #29172
Fixes #28969

### CODE REVIEW

File: types/node/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for Node.js 10.11
+// Type definitions for Node.js 10.11.x
```
Comment (andy-ms) [inline types/node/index.d.ts]: All other changes were to remove `.x`, why adding it here? That should violate `dt-header`.

**Comment by andy-ms** [comment]
Issue already fixed in the publisher but lint fixes are nice to have anyway.

**Comment by SimonSchick** [comment]
@andy-ms I amended the Set change into this PR.

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**